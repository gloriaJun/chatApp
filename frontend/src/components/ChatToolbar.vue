<template>
  <v-toolbar
    class="chat-toolbar"
    color="amber darken-3"
    height="50px"
    dark
  >

    <slot name="prepend"></slot>

    <v-toolbar-title
      class="text-xs-center"
    >
      {{ title }}
    </v-toolbar-title>

    <v-spacer></v-spacer>

    <v-btn
      icon
      @click="search"
    >
      <v-icon>search</v-icon>
    </v-btn>

    <slot name="append"></slot>

  </v-toolbar>
</template>

<script>
export default {
  name: 'ChatToolbar',
  props: {
    title: {
      type: String,
      default: '',
    },
  },
  data: () => ({
  }),
  methods: {
    search() {
      this.$emit('search');
    },
  },
};
</script>
