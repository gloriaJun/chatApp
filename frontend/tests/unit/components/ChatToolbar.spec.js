import {
  mount,
} from '@vue/test-utils';
import '../test-setup';

import ChatToolbar from '@/components/ChatToolbar.vue';

describe('ChatToolbar.vue', () => {
  it('snapshot check', () => {
    const wrapper = mount(ChatToolbar);

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('event emit - search', () => {
    const wrapper = mount(ChatToolbar);

    wrapper.vm.search()

    expect(wrapper.emitted().search[0]).toEqual([]);
  });

  it('props - title', () => {
    const title = 'ChatRoom';
    const wrapper = mount(ChatToolbar, {
      propsData: { title },
    });

    expect(wrapper.find('.v-toolbar__title').text()).toBe(title);
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
