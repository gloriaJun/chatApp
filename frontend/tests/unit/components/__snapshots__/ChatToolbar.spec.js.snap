// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatToolbar.vue props - title 1`] = `
<nav
  class="chat-toolbar v-toolbar theme--dark amber darken-3"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
>
  <div
    class="v-toolbar__content"
    style="height: 50px;"
  >
     
    <div
      class="v-toolbar__title text-xs-center"
    >
      
    ChatRoom
  
    </div>
     
    <div
      class="spacer"
    />
     
    <button
      class="v-btn v-btn--icon theme--dark"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--dark"
        >
          search
        </i>
      </div>
    </button>
     
  </div>
</nav>
`;

exports[`ChatToolbar.vue snapshot check 1`] = `
<nav
  class="chat-toolbar v-toolbar theme--dark amber darken-3"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
>
  <div
    class="v-toolbar__content"
    style="height: 50px;"
  >
     
    <div
      class="v-toolbar__title text-xs-center"
    >
      
    
  
    </div>
     
    <div
      class="spacer"
    />
     
    <button
      class="v-btn v-btn--icon theme--dark"
      type="button"
    >
      <div
        class="v-btn__content"
      >
        <i
          aria-hidden="true"
          class="v-icon material-icons theme--dark"
        >
          search
        </i>
      </div>
    </button>
     
  </div>
</nav>
`;
